pkgs/applications/science/logic/bitwuzla/default.nix

```diff
@@ -0,0 +1,66 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+, python3
+, cmake
+, lingeling
+, btor2tools
+, gtest
+, gmp
+, cadical
+, minisat
+, picosat
+, cryptominisat
+, zlib
+, pkg-config
+}:
+
+stdenv.mkDerivation rec {
+  pname = "bitwuzla";
+  version = "unstable-2021-07-01";
+
+  src = fetchFromGitHub {
+    owner = "bitwuzla";
+    repo = "bitwuzla";
+    rev = "58d720598e359b1fdfec4a469c76f1d1f24db51a";
+    sha256 = "06ymqsdppyixb918161rmbgqvbnarj4nm4az88lkn3ri4gyimw04";
+  };
+
+  nativeBuildInputs = [ cmake pkg-config ];
+  buildInputs = [
+    lingeling
+    cadical
+    cryptominisat
+    picosat
+    minisat
+    btor2tools
+    gmp
+    zlib
+  ];
+
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=ON"
+    "-DPicoSAT_INCLUDE_DIR=${lib.getDev picosat}/include/picosat"
+    "-DBtor2Tools_INCLUDE_DIR=${lib.getDev btor2tools}/include/btor2parser"
+    "-DBtor2Tools_LIBRARIES=${lib.getLib btor2tools}/lib/libbtor2parser.so"
```

Comment:
On Darwin, I got
```
make[2]: *** No rule to make target '/nix/store/yzn2n7pmf4iykqagq65nznm0jmyyqap8-btor2tools-1.0.0-pre_9831f9909fb283752a3d6d60d43613173bd8af42-lib/lib/libbtor2parser.so', needed by 'lib/libbitwuzla.dylib'.  Stop.                 
```

So this would be something like:
```
(if stdenv.isDarwin then "-DBtor2Tools_LIBRARIES=${lib.getLib btor2tools}/lib/libbtor2parser.dylib" else ...)
```